packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -495,11 +441,31 @@ export function useConnections({
   }, [getAutoConnectInfo, persistOIDCTokens]);
 
   useEffectOnChange(() => {
-    if (!persistOIDCTokens)
-      void loadConnections(dispatch, connectionRepository, {
+    void loadConnections(
+      dispatch,
+      storedFavoriteConnections,
+      storedNonFavoriteConnections,
+      {
         persistOIDCTokens,
-      });
-  }, [persistOIDCTokens]);
+      }
+    );
+  }, [storedFavoriteConnections, storedNonFavoriteConnections]);
+
+  useEffectOnChange(() => {
+    if (!persistOIDCTokens)
+      void loadConnections(
+        dispatch,
+        storedFavoriteConnections,
+        storedNonFavoriteConnections,
+        {
+          persistOIDCTokens,
+        }
+      );
+  }, [
+    storedFavoriteConnections,
+    storedNonFavoriteConnections,
+    persistOIDCTokens,
+  ]);
```

Comment:
Doesn't look like we need two hooks here and this should probably also run on the initial mount, right? Any reason we don't want internal state of `connectionsReducer` to be as much in sync as possible with the repository? 🤔 Looks like this should be just one `useEffect` hook that depends on all three values and updates the reducer hook state whenever any of those other values change?

Reply:
Actually we want it in sync as much as possible. Maybe useEffect and removing the if so we always sync is good enough? Let me try. I tried once before doing this but likely did it wrong.

Reply:
Hmmm, I was taking a closer look, and two things caught my eye:

1) in terms of this state being inside the hook, it seems like we don't need it there anymore and can just return the one from the repository hook directly.
2) seems like when `persistOIDCTokens` is changed, we should actually update stored connections, I don't see this happening, is it possible that this is a regression? Maybe I'm just missing the place where this happens

Reply:
There is even this `toBeReSaved` array that gets values pushed, but is never used to actually save stuff

Reply:
Yeah I fixed the toBeResaved, because I was thinking on removing it, but I realised it was necessary, because we may have OIDC tokens but they don't need to be persisted, so we need some "intermediate in memory stage" that we can flush to disk without changes. Maybe there is another way, we can discuss alternatives in the sync because maybe there is something simple that we can just do and everything works.